### ISSUE TITLE
[BUG] The test var_ode_sys fails

### ISSUE THREAD

**Issue by yurivict** [bluescarni/heyoka#430]
```
123/140 Testing: var_ode_sys
123/140 Test: var_ode_sys
Command: "/usr/ports/math/heyoka/work/.build/test/var_ode_sys"
Directory: /usr/ports/math/heyoka/work/.build/test
"var_ode_sys" start time: Jun 14 10:35 PDT
Output:
----------------------------------------------------------

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
var_ode_sys is a Catch v2.13.10 host application.
Run with -? for options

-------------------------------------------------------------------------------
nonauto sys
-------------------------------------------------------------------------------
/usr/ports/math/heyoka/work/heyoka-5.0.0/test/var_ode_sys.cpp:408
...............................................................................

/usr/ports/math/heyoka/work/heyoka-5.0.0/test/var_ode_sys.cpp:467: FAILED:
  REQUIRE( ta_orig.get_state()[0] == approximately(ta.get_state()[0] + ta.get_state()[4] * delta_par + ta.get_state()[5] * delta_tm) )
with expansion:
  0.0258427687 == 0.02584276874817543

===============================================================================
test cases:   6 |   5 passed | 1 failed
assertions: 138 | 137 passed | 1 failed

<end of output>
Test time =   5.21 sec
----------------------------------------------------------
Test Failed.
```

Version: 5.0.0 + PR patches
FreeBSD 14.1
Clang-16

**Comment by bluescarni** [comment]
@yurivict thanks for the reports, I have updated #428 with relaxed tolerances for the failing numerical tests. This includes also tentative fixes for #429 . Would you mind checking it out?

I would like to add a FreeBSD build to the CI in order to catch these sort of issues before releases. Do you have any experience using this github action:

https://github.com/marketplace/actions/freebsd-vm

And/or, do you have any suggestions on how to set up a FreeBSD CI pipeline on github/circleCI/similar?

**Comment by bluescarni** [comment]
@yurivict did you have a chance to test the fixes in #428?

**Comment by bluescarni** [comment]
Closing as I am assuming this is fixed, please re-open if needed.
